source/common/upstream/upstream_impl.cc

```diff
@@ -302,7 +302,7 @@ std::vector<HostsPerLocalityConstSharedPtr> HostsPerLocalityImpl::filter(
     }
 
     for (size_t i = 0; i < predicates.size(); ++i) {
-      mutable_clones[i]->hosts_per_locality_.push_back(std::move(current_locality_hosts[0]));
+      mutable_clones[i]->hosts_per_locality_.push_back(std::move(current_locality_hosts[i]));
```

Comment:
Is this a previous bug unrelated to this change? What's the implication?

Reply:
Previous bug yeah. I think this means that degraded locality routing does not work correctly when there are more than one locality, so it's pretty bad but localized to a very specific situation.